### DISCUSSION TITLE
Listview Enhancements Suggestions

### DISCUSSION THREAD

**Discussion by hfloyd** [umbraco/Umbraco-CMS#12393] [Features and ideas]
I really like the Listviews, but there are a few things that always annoy me..

1 - The required "Published" column:
![image](https://user-images.githubusercontent.com/1252593/167911231-c336c592-d9e5-47f8-bf92-a5b762226004.png)
IMO this adds no value - unpublished nodes are shown greyed-out already:
![image](https://user-images.githubusercontent.com/1252593/167910973-5c76686e-af20-4541-bcab-79319d01dbe8.png)
The width of the column takes up space better used by useful property data. It would be great if this were treated as a "regular" column type and could be removed from the customized Listview - like the "sortOrder" or "updateDate" columns.

2 - I wish the column widths could be somehow configured (maybe optionally, using %s?) Often I have a longer (useful) textstring property I'd like displayed, but other properties take up so much unnecessary space (example: Sort Order is really wide):
![image](https://user-images.githubusercontent.com/1252593/167912077-3e446ca6-e09a-4366-b3fd-0bd220495bbc.png)

**Comment by nathanwoulfe** [comment]
The column widths always bugs me too - I have a feeling it's because the view is built to look like a table, rather than actually using table markup. Caveat - haven't looked at the code, this is from memory, which might be wrong 🙄

If it were a table, the browser would flow the column widths to best-fit the content, but as it's a stack of divs, the widths are set explicitly(or via flex rules). 

Same issue goes for the umb-table directive, which should really be named umb-lotsa-divs-looking-like-a-table, but that's a bit much.

**Reply by hfloyd** [reply]
It's like after those old days of web dev where people would design websites using photoshop and a mess of nested tables to hold it together, we've now shunned tables - even in places where they SHOULD be used ;-)

I'd be happy with an extra option on each property (like there are options for the "Display name" and "template") to add a width value of some type. I'd assume that could be used even with collections of divs.

**Comment by hfloyd** [comment]
So today I came across another listview annoyance, That "Status" column (which can't be removed) is not even sortable, so it really has ZERO value there in the view. I'd like it to be sortable AND removable.,
